addons/l10n_latam_check/models/account_payment.py

```diff
@@ -0,0 +1,283 @@
+    l10n_latam_use_checkbooks = fields.Boolean(related='journal_id.l10n_latam_use_checkbooks')
+    l10n_latam_checkbook_type = fields.Selection(related='l10n_latam_checkbook_id.type')
+    l10n_latam_checkbook_id = fields.Many2one(
+        'l10n_latam.checkbook', 'Checkbook', store=True,
+        compute='_compute_l10n_latam_checkbook', readonly=True, states={'draft': [('readonly', False)]})
+
+    @api.depends('payment_method_line_id.code', 'journal_id.l10n_latam_use_checkbooks')
+    def _compute_l10n_latam_checkbook(self):
+        with_checkbooks = self.filtered(lambda x: x.payment_method_line_id.code == 'check_printing' and
+                                                  x.journal_id.l10n_latam_use_checkbooks)
+        (self - with_checkbooks).l10n_latam_checkbook_id = False
+        for rec in with_checkbooks:
+            checkbooks = rec.journal_id.l10n_latam_checkbook_ids
+            if rec.l10n_latam_checkbook_id and rec.l10n_latam_checkbook_id in checkbooks:
+                continue
+            rec.l10n_latam_checkbook_id = checkbooks and checkbooks[0] or False
+
+    @api.depends('l10n_latam_checkbook_id', 'journal_id', 'payment_method_code')
+    def _compute_check_number(self):
+        """ Override from account_check_printing"""
+        from_checkbooks = self.filtered(lambda x: x.l10n_latam_checkbook_id)
+        for pay in from_checkbooks:
+            pay.check_number = pay.l10n_latam_checkbook_id.sequence_id.get_next_char(
+                pay.l10n_latam_checkbook_id.next_number)
+        return super(AccountPayment, self - from_checkbooks)._compute_check_number()
+
+    def action_unmark_sent(self):
+        """ Check that the recordset is valid, set the payments state to sent and call print_checks() """
+        if self.filtered('l10n_latam_checkbook_id'):
+            raise UserError(_('Unmark sent is not implemented for checks that use checkbooks'))
+
+    @api.onchange('l10n_latam_check_id')
+    def _onchange_check(self):
+        for rec in self.filtered('l10n_latam_check_id'):
+            rec.amount = rec.l10n_latam_check_id.amount
+
+    @api.depends('payment_method_line_id.code', 'partner_id')
+    def _compute_l10n_latam_check_data(self):
+        new_third_party_checks = self.filtered(lambda x: x.payment_method_line_id.code == 'new_third_party_checks')
+        for rec in new_third_party_checks:
+            rec.update({
+                'l10n_latam_check_bank_id': rec.partner_id.bank_ids and rec.partner_id.bank_ids[0].bank_id or False,
+                'l10n_latam_check_issuer_vat': rec.partner_id.vat,
+            })
+
+    @api.depends(
+        'payment_method_line_id', 'l10n_latam_check_issuer_vat', 'l10n_latam_check_bank_id', 'company_id',
+        'check_number', 'l10n_latam_check_id', 'state', 'date', 'is_internal_transfer')
+    def _compute_l10n_latam_check_warning_msg(self):
+        self.l10n_latam_check_warning_msg = False
+        for rec in self.filtered(lambda x: x.state == 'draft'):
```

Comment:
mm, the for should iterate over more records than the one with latam_check id. This is already taked into account on the if inside. 
If we want this on the for we should do something like this:
```
for rec in self.filtered(lambda x: x.state == 'draft' and (x.l10n_latam_check_id or x.payment_method_line_id.code == 'new_third_party_checks' or x.l10n_latam_checkbook_id)):
```

maybe it could be something like this
```

latam_draft_checks = self.filtered(lambda x: x.state == 'draft' and (x.payment_method_line_id.code in ['in_third_party_checks', 'out_third_party_checks'] or x.l10n_latam_checkbook_id))
for rec in latam_draft_checks:


```

Reply:
better formatting done